lib/watir-webdriver/locators/element_locator.rb

```diff
@@ -146,6 +159,15 @@ def wd_find_by_regexp_selector(selector, method = :find)
         raise Error, "internal error: unable to build WebDriver selector from #{selector.inspect}"
       end
 
+      if how == :xpath && can_convert_regexp_to_contains?
+        rx_selector.each do |key, value|
+          unless key == :tag_name
```

Comment:
Sounds good. I had originally omitted conversion for text-node matching as well. I don't remember if that was why, but it's simple enough to reimplement.

As for performance increases, I suspect you'll see the most gain when driving a browser across higher latency networks, at least that's where [our impetus](https://phabricator.wikimedia.org/T92613) for making this optimization came from.